### PR TITLE
[wmco] Handle errors in Machine deletion check

### PR DESCRIPTION
**Pull request by sebsoto**
This commit makes it so that the isAllowedDeletion() function properly
returns an error in the case that the function is unable to tell if the
Machine can be deleted or not.

### CODE REVIEW

File: controllers/windowsmachine_controller.go [hunk 1]
```diff
@@ -448,31 +452,31 @@ func (r *WindowsMachineReconciler) validateUserData(privateKey []byte) error {
 
 // isAllowedDeletion determines if the number of machines after deletion of the given machine doesn`t fall below the
 // minHealthyCount
-func (r *WindowsMachineReconciler) isAllowedDeletion(machine *mapi.Machine) bool {
+func (r *WindowsMachineReconciler) isAllowedDeletion(machine *mapi.Machine) (bool, error) {
 	if len(machine.OwnerReferences) == 0 {
-		return false
+		return false, fmt.Errorf("machine has no owner reference")
```
Comment (mansikulkarni96) [inline controllers/windowsmachine_controller.go:457]: can use errors.New here?

File: controllers/windowsmachine_controller.go [hunk 2]
```diff
@@ -448,31 +452,31 @@ func (r *WindowsMachineReconciler) validateUserData(privateKey []byte) error {
...
 	}
 	machinesetName := machine.OwnerReferences[0].Name
 
 	machines := &mapi.MachineList{}
 	err := r.client.List(context.TODO(), machines,
 		client.MatchingLabels(map[string]string{MachineOSLabel: "Windows"}))
 	if err != nil {
-		return false
+		return false, errors.Wrap(err, "cannot list machines")
```
Comment (aravindhp) [inline controllers/windowsmachine_controller.go:465]: `Machines`

File: controllers/windowsmachine_controller.go (hunk 1 above)
Comment (aravindhp) [reply controllers/windowsmachine_controller.go:457]: +1 to `errors.New` and please use `Machines`

**Review by aravindhp — APPROVED** [review]
Thanks for fixing this, @sebsoto 

With the minor fixes in, I
/approve

**Comment by mansikulkarni96** [comment]
Thanks for working on this @sebsoto 
/lgtm

**Comment by sebsoto** [comment]
/hold
while network test issues persist

**Pull request merged**